getting_started/kubernetes-overview.adoc

```diff
@@ -0,0 +1,105 @@
+
+|`kube-proxy`
+|Runs on every node in the cluster and maintains the network traffic between the Kubernetes resources.
+
+|`kube-controller-manager`
+|Governs the state of the cluster.
+
+|`scheduler`
+|Allocates pods to nodes.
+
+|`etcd`
+|Stores cluster data.
+
+|`kube-api-server`
+|Validates and configures data for the API objects.
+
+|`kubelet`
+|Runs on nodes and reads the container manifests. Ensures that the defined containers have started and are running.
+
+|`kubectl`
+|Allows you to define how you want to run workloads. Use the `kubectl` command to interact with the `kube-api-server`.
+
+|Node
+|Node is a physical or VM in a Kubernetes cluster. The control plane manages every node and schedules pods across the nodes in the Kubernetes cluster.
+
+|`container-runtime`
+|`container-runtime` runs containers on a host operating system. You must install container-runtime on each node so that pods can run on the node.
+
+|Persistent storage
+|Stores the data even after the device is shut  off. Kubernetes uses persistent volumes to store the application data.
+
+|`container-registry`
+|Stores and accesses the container images.
+
+|Pod
+|The pod is the smallest logical unit in Kubernetes. A pod contains one or more containers to run in a worker node.
+|===
+
+A cluster is a single computational unit consisting of multiple nodes in a cloud environment. A Kubernetes cluster includes a control plane and worker nodes. You can run Kubernetes containers across various machines and environments. The control plane node controls and maintains the state of a cluster. You can run the Kubernetes application by using worker nodes. You can use the Kubernetes namespace to differentiate cluster resources in a cluster. Namespace scoping is applicable for resource objects, such as deployment, service, and pods. You can not use namespace for cluster-wide resource objects such as `storageclass`, nodes, and persistent volumes.
+
+.Kubernetes cluster overview
+image::247_OpenShift_Kubernetes_Overview-1.png[]
+
+A custom resource is an extension of the Kubernetes API. You can customize Kubernetes clusters by using custom resources. Operators are software extensions which manage applications and their components with the help of custom resources. Kubernetes uses a declarative model when you want a fixed desired result while dealing with cluster resources. By using Operators, Kubernetes defines its states in a declarative way. You can modify the Kubernetes cluster resources by using imperative commands.
+An Operator acts as a control loop which continuously compares the desired state of resources with the actual state of resources and puts actions in place to bring reality in line with the desired state.
+
+.Kubernetes Resources
+[cols="1,2",options="header"]
+|===
+|Resource |Purpose
+
+|Service
+|Kubernetes uses services to expose a running application on a set of pods.
+
+|ReplicaSets
+|Kubernetes uses ReplicaSets to maintain the constant pod number.
```

Comment:
Kubernetes uses the `ReplicaSets` resource

Reply:
Done